From: Miroslav Benes <mbenes@suse.cz>
Subject: [PATCH] kgraft/bnx2fc: Do not block kGraft in bnx2fc_l2_rcv kthread
Patch-mainline: not yet, kGraft
References: bsc#1094033, fate#313296

kthread loop in bnx2fc_l2_rcv_thread() is not marked with
klp_kgraft_mark_task_safe() and as such it can block the patching
process indefinitely. Mark it.

Signed-off-by: Miroslav Benes <mbenes@suse.cz>
---
 drivers/scsi/bnx2fc/bnx2fc_fcoe.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/scsi/bnx2fc/bnx2fc_fcoe.c
+++ b/drivers/scsi/bnx2fc/bnx2fc_fcoe.c
@@ -475,6 +475,7 @@ static int bnx2fc_l2_rcv_thread(void *ar
 	set_current_state(TASK_INTERRUPTIBLE);
 	while (!kthread_should_stop()) {
 		schedule();
+		klp_kgraft_mark_task_safe(current);
 		spin_lock_bh(&bg->fcoe_rx_list.lock);
 		while ((skb = __skb_dequeue(&bg->fcoe_rx_list)) != NULL) {
 			spin_unlock_bh(&bg->fcoe_rx_list.lock);
